### PR TITLE
revised check_message_overflow to split messages into many smaller me…

### PR DESCRIPTION
**Pull request by KJStick**
…ssages

### CODE REVIEW

File: wxt_cidrbot/cidrbot.py [hunk 1]
```diff
@@ -172,31 +172,70 @@ def webhook_request(self, event):
...
+                #if end message, don't do split_keyword part
+                message_first_part = split_keyword + repo_name + " (continued)" + "\n" + repo_part
+        msgs_to_be_sent.append(message_first_part)
 
         if request_type == 'daily_message':
-            self.send_wbx_msg(room_id, message_first_part, pt_id)
-        if request_type == 'edit_message':
-            self.webex.edit_message(message_id, message_first_part, room_id)
-        self.send_wbx_msg(room_id, remainder_message, pt_id)
+            self.send_wbx_msg(room_id, msgs_to_be_sent.pop(0), pt_id)
```
Comment (ppajersk) [inline wxt_cidrbot/cidrbot.py:233]: To reduce code complexity, you can split the function here, and delegate to another function to send messages and return the true

File: wxt_cidrbot/cidrbot.py [hunk 2]
```diff
@@ -172,31 +172,70 @@ def webhook_request(self, event):
...
+
+            split_repo = repo.split("\n")
+            repo_name = split_repo.pop(0)
+
+            #check to see if cannot append at least n lines to last message
+            is_not_able_to_append = len(message_first_part) + len(
+                "".join(split_repo[:num_needed_repo_lines])
+            ) >= self.WEBEX_MESSAGE_CHAR_LIMIT
+
+            #check see if entire repo fit in one message so don't waste memory and time splitting and repiecing
```
Comment (ppajersk) [inline wxt_cidrbot/cidrbot.py:213]: Can this functionality be it's own Boolean function?

**Review by ppajersk — COMMENTED** [review]
Not ready to merge yet. Check comments, get Josh's review, and fix the sonarcloud complexity

**Review by ppajersk — APPROVED** [review]
Looks and works good

File: wxt_cidrbot/cidrbot.py [hunk 3]
```diff
@@ -51,11 +51,33 @@ def __init__(self):
...
+
+        return: bool, True if messages sent
+        """
+
+        if request_type == 'daily_message':
+            self.send_wbx_msg(room_id, msgs_to_be_sent.pop(0), pt_id)
+        elif request_type == 'edit_message':
+            self.webex.edit_message(message_id, msgs_to_be_sent.pop(0), room_id)
+        for msg in msgs_to_be_sent:
+            self.send_wbx_msg(room_id, msg, pt_id)
```
Comment (jlothian) [inline wxt_cidrbot/cidrbot.py:77]: I'm not sure I get how this works - the first message in a list will be either a daily message or an edit, but all the rest will be sent the same way?

**Review by jlothian — APPROVED** [review]

**Pull request merged**